src/EntityRegistrantDetection/CallbackWrapper.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * Class CallbackWrapper.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\EntityRegistrantDetection;
+
+use AmpProject\AmpWP\Services;
+use ArrayAccess;
+use WP_Block_Type_Registry;
+
+/**
+ * Callback wrapper for EntityRegistrantDetection.
+ *
+ * @since   2.2
+ * @package AmpProject\AmpWP
+ * @internal
+ */
+class CallbackWrapper implements ArrayAccess {
+
+	/**
+	 * List of default entities.
+	 *
+	 * @var array [
+	 *     @type array $post_type List of default post types slug.
+	 *     @type array $taxonomy  List of default taxonomy slug.
+	 *     @type array $block     List of default block type.
+	 *     @type array $shortcode List of default shortcode.
+	 * ]
+	 */
+	const DEFAULT_ENTITIES = [
```

Comment:
Instead of storing this here, I think it would be better to have a `$initial_entities` variable which is `null` to begin with, then when the callback wrapper is invoked it can check if `$this->initial_entities === null`, and of so, populate the initial array with what is registered. 

Nevertheless, I wonder if it is necessary? If we capture all of the registered entities at `prepare` and then diff them with what is registered at `finalize`, why do we need to also need the default entities?

Reply:
> Instead of storing this here, I think it would be better to have a $initial_entities variable which is null to begin with, then when the callback wrapper is invoked it can check if $this->initial_entities === null, and of so, populate the initial array with what is registered.

Currently, We are doing the same but with `$registered_entities`. In `prepare()`, we collect all registered entities and in `finalize()` we update the same property with a difference of registered entities.

> Nevertheless, I wonder if it is necessary? If we capture all of the registered entities at prepare and then diff them with what is registered at `finalize()`, why do we need to also need the default entities?

Since We are checking all the callbacks from all hooks. It will also track the WordPress default functions which registers default entities. which I think is unnecessary to track. (Since the idea to detect if any theme/plugin is registering entities or not) and will increase the REST API payload.
So to exclude WordPress default entities in `finalize()` this const is being used.

Reply:
Sorry I haven't been able to respond to this yet. I'll have to dig more deeply into the PR and its changes in order to be able to reply. That'll have to come next week.